### ISSUE TITLE
Compilation error in quickstart example

### ISSUE THREAD

**Issue by sandeep-ps** [mlpack/mlpack#3367]
#### Problem location

https://github.com/mlpack/mlpack/blob/master/doc/quickstart/cpp.md#simple-quickstart-example

#### Description of problem

There is an error when compiling cpp_quickstart_1.cpp:

```
cpp_quickstart_1.cpp: In function 'int main()':
cpp_quickstart_1.cpp:32:16: error: missing template arguments before 'r'
   32 |   RandomForest r(trainDataset,
      |                ^
cpp_quickstart_1.cpp:40:3: error: 'r' was not declared in this scope
   40 |   r.Classify(trainDataset, trainPredictions);
      |   ^
```
mlpack versions: 4.0.0 and 4.0.1

I tried used mlpack 4.0.1 and 4.0.0 docker containers using `docker run -it mlpack/mlpack /bin/bash` and `docker run -it mlpack/mlpack:4.0.0 /bin/bash` commands respectively.  I also saw a related issue that was closed here: https://github.com/mlpack/mlpack/issues/3340

I'm creating this issue as part of JOSS review for mlpack 4.0.0 https://github.com/openjournals/joss-reviews/issues/5026. Thanks.

**Comment by sandeep-ps** [comment]
The compilation goes away after changing `RandomForest` to `RandomForest<>`, as mentioned in https://github.com/mlpack/mlpack/pull/3341. I'm pretty sure, I'm using mlpack 4.0.0. So, it's clear where the problem lies.

**Comment by rcurtin** [comment]
Thanks @sandeep-ps!  You are indeed correct.  This is just a typo in the quickstart.  I opened PR #3373 to fix the issue. :+1:

**Comment by conradsnicta** [comment]
the fix in PR #3373 has been merged

**Comment by sandeep-ps** [comment]
Thank you!
